File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> sa(nums1.begin(),nums1.end());

        vector<int>ans;

        for(int i = 0 ; i<nums2.size();i++){
            if(sa.find(nums2[i])!=sa.end()){
                ans.push_back(nums2[i]);
                sa.erase(nums2[i]);
            }
        }

        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let s = new Set();

    for(const a of nums1){
        s.add(a);
    }

    let ans = [];

    for(const a of nums2){
        if(s.has(a)){
            ans.push(a);
            s.delete(a);
        }
    }
    return ans;
};